engine/renderer/private/pipelines/shadow_pipeline.cpp

```diff
@@ -60,62 +69,65 @@ void ShadowPipeline::RecordCommands(vk::CommandBuffer commandBuffer, uint32_t cu
         .pDepthAttachment = &depthAttachmentInfo,
     };
 
-    commandBuffer.beginRenderingKHR(&renderingInfo, _brain.dldi);
+    commandBuffer.beginRenderingKHR(&renderingInfo, _context->VulkanContext()->Dldi());
 
     commandBuffer.bindPipeline(vk::PipelineBindPoint::eGraphics, _pipeline);
 
     commandBuffer.bindDescriptorSets(vk::PipelineBindPoint::eGraphics, _pipelineLayout, 0, { scene.gpuScene->GetObjectInstancesDescriptorSet(currentFrame) }, {});
     commandBuffer.bindDescriptorSets(vk::PipelineBindPoint::eGraphics, _pipelineLayout, 1, { scene.gpuScene->GetSceneDescriptorSet(currentFrame) }, {});
 
-    vk::Buffer vertexBuffer = _brain.GetBufferResourceManager().Access(scene.batchBuffer->VertexBuffer())->buffer;
-    vk::Buffer indexBuffer = _brain.GetBufferResourceManager().Access(scene.batchBuffer->IndexBuffer())->buffer;
-    vk::Buffer indirectDrawBuffer = _brain.GetBufferResourceManager().Access(_drawBuffer)->buffer;
-    vk::Buffer indirectCountBuffer = _brain.GetBufferResourceManager().Access(scene.gpuScene->IndirectCountBuffer(currentFrame))->buffer;
+    vk::Buffer vertexBuffer = resources->BufferResourceManager().Access(scene.batchBuffer->VertexBuffer())->buffer;
+    vk::Buffer indexBuffer = resources->BufferResourceManager().Access(scene.batchBuffer->IndexBuffer())->buffer;
+    vk::Buffer indirectDrawBuffer = resources->BufferResourceManager().Access(_drawBuffer)->buffer;
+    vk::Buffer indirectCountBuffer = resources->BufferResourceManager().Access(scene.gpuScene->IndirectCountBuffer(currentFrame))->buffer;
 
     commandBuffer.bindVertexBuffers(0, { vertexBuffer }, { 0 });
     commandBuffer.bindIndexBuffer(indexBuffer, 0, scene.batchBuffer->IndexType());
+
     uint32_t indirectCountOffset = scene.gpuScene->IndirectCountOffset();
-    commandBuffer.drawIndexedIndirectCountKHR(indirectDrawBuffer, 0, indirectCountBuffer, indirectCountOffset, scene.gpuScene->DrawCount(), sizeof(vk::DrawIndexedIndirectCommand), _brain.dldi);
-    _brain.drawStats.drawCalls++;
-    _brain.drawStats.indirectDrawCommands += scene.gpuScene->DrawCount();
-    _brain.drawStats.indexCount += scene.gpuScene->DrawCommandIndexCount();
+    commandBuffer.drawIndexedIndirectCountKHR(indirectDrawBuffer, 0, indirectCountBuffer, indirectCountOffset, scene.gpuScene->DrawCount(), sizeof(vk::DrawIndexedIndirectCommand), vkContext->Dldi());
 
-    commandBuffer.endRenderingKHR(_brain.dldi);
+    _context->GetDrawStats().IndirectDraw(scene.gpuScene->DrawCount(), scene.gpuScene->DrawCommandIndexCount());
```

Comment:
With a command list abstraction, we can track all of these draw statistic much more easily over every single command list and then sum them at the end of a frame for an accurate result (it is possible that someone forgets to update the draw stats after issueing a draw command)